XL: GetObject: Crash when XL.GetObject is called with negative offset.

## Expected behaviour

Not sure of expected behavior when offset is negative, but it shouldn't crash atleast. 
## Actual behaviour

Crashes with following stack trace. 

```
=== RUN   TestGetObject
--- FAIL: TestGetObject (0.14s)
    object-api-getobject_test.go:128: Test 1: FS: Expected to fail with error "The requested range is not satisfiable", but instead failed with error "seek /var/folders/wz/2sl5qjts3cv9zcsjk7tvp5wr0000gn/T/minio-127883649/yhimqymqjnnvkv1hyw5nnzkafljq343xz6h9qnfmyqr4sr0pk2tm2v4jyqmw/test-object: invalid argument" instead
panic: runtime error: slice bounds out of range [recovered]
    panic: runtime error: slice bounds out of range

goroutine 18 [running]:
panic(0x748ae0, 0xc820010060)
    /usr/local/go/src/runtime/panic.go:464 +0x3e6
testing.tRunner.func1(0xc8201a0870)
    /usr/local/go/src/testing/testing.go:467 +0x192
panic(0x748ae0, 0xc820010060)
    /usr/local/go/src/runtime/panic.go:426 +0x4e9
github.com/minio/minio.writeDataBlocks(0x1a998f0, 0xc8201f2560, 0xc8201d1500, 0x10, 0x10, 0x8, 0xffffffffffffffff, 0x600000, 0x10, 0x0, ...)
    /Users/hackintoshrao/mycode/go/src/github.com/minio/minio/erasure-utils.go:102 +0x517
github.com/minio/minio.erasureReadFile(0x1a998f0, 0xc8201f2560, 0xc82019a200, 0x10, 0x10, 0xc82013adc0, 0x3c, 0xc82017c760, 0x12, 0xc8200108f0, ...)
    /Users/hackintoshrao/mycode/go/src/github.com/minio/minio/erasure-readfile.go:293 +0x77f
github.com/minio/minio.xlObjects.GetObject(0xc82019a100, 0x10, 0x10, 0xc8200ba400, 0x10, 0x10, 0x8, 0x8, 0x9, 0x9, ...)
    /Users/hackintoshrao/mycode/go/src/github.com/minio/minio/xl-v1-object.go:154 +0x1660
github.com/minio/minio.(*xlObjects).GetObject(0xc8201c0070, 0xc82013adc0, 0x3c, 0x83b720, 0xb, 0xffffffffffffffff, 0x600000, 0x1a998f0, 0xc8201f2560, 0x0, ...)
    <autogenerated>:250 +0x136
github.com/minio/minio.testGetObject(0x1a531f8, 0xc8201c0070, 0x818de0, 0x2, 0xc8201a0870)
    /Users/hackintoshrao/mycode/go/src/github.com/minio/minio/object-api-getobject_test.go:118 +0xc37
github.com/minio/minio.ExecObjectLayerTest(0xc8201a0870, 0x979dd8)
    /Users/hackintoshrao/mycode/go/src/github.com/minio/minio/test-utils_test.go:653 +0x319
github.com/minio/minio.TestGetObject(0xc8201a0870)
    /Users/hackintoshrao/mycode/go/src/github.com/minio/minio/object-api-getobject_test.go:29 +0x2d
testing.tRunner(0xc8201a0870, 0xc89178)
    /usr/local/go/src/testing/testing.go:473 +0x98
created by testing.RunTests
    /usr/local/go/src/testing/testing.go:582 +0x892
exit status 2
```
## Steps to reproduce the behaviour

run the test here https://github.com/hackintoshrao/minio/blob/9fa75bc1156155ca2c409aea3858355833b31b0a/object-api-getobject_test.go
## Minio version

Current master at this point in time 169c72cdab14ac0f993161f6ed97e8462eccfb97 . 
## System information

OSX ElCaptain.

--- a/erasure-readfile.go
+++ b/erasure-readfile.go
@@ -175,6 +175,11 @@ func parallelRead(volume, path string, readDisks []StorageAPI, orderedDisks []St
 // then written to given writer. This function also supports bit-rot detection by
 // verifying checksum of individual block's checksum.
 func erasureReadFile(writer io.Writer, disks []StorageAPI, volume string, path string, partName string, eInfos []erasureInfo, offset int64, length int64, totalLength int64) (int64, error) {
+	// Offset and length cannot be negative.
+	if offset < 0 || length < 0 {
+		return 0, errUnexpected
+	}
+
 	// Pick one erasure info.
 	eInfo := pickValidErasureInfo(eInfos)
 

--- a/erasure-utils.go
+++ b/erasure-utils.go
@@ -74,6 +74,11 @@ func getDataBlockLen(enBlocks [][]byte, dataBlocks int) int {
 // Writes all the data blocks from encoded blocks until requested
 // outSize length. Provides a way to skip bytes until the offset.
 func writeDataBlocks(dst io.Writer, enBlocks [][]byte, dataBlocks int, outOffset int64, outSize int64) (int64, error) {
+	// Offset and out size cannot be negative.
+	if outOffset < 0 || outSize < 0 {
+		return 0, errUnexpected
+	}
+
 	// Do we have enough blocks?
 	if len(enBlocks) < dataBlocks {
 		return 0, reedsolomon.ErrTooFewShards

--- a/fs-v1.go
+++ b/fs-v1.go
@@ -217,6 +217,10 @@ func (fs fsObjects) GetObject(bucket, object string, offset int64, length int64,
 	if !IsValidObjectName(object) {
 		return ObjectNameInvalid{Bucket: bucket, Object: object}
 	}
+	// Offset and length cannot be negative.
+	if offset < 0 || length < 0 {
+		return toObjectErr(errUnexpected, bucket, object)
+	}
 	var totalLeft = length
 	bufSize := int64(readSizeV1)
 	if length > 0 && bufSize > length {

--- a/xl-v1-object.go
+++ b/xl-v1-object.go
@@ -48,7 +48,10 @@ func (xl xlObjects) GetObject(bucket, object string, startOffset int64, length i
 	if !IsValidObjectName(object) {
 		return ObjectNameInvalid{Bucket: bucket, Object: object}
 	}
-
+	// Start offset and length cannot be negative.
+	if startOffset < 0 || length < 0 {
+		return toObjectErr(errUnexpected, bucket, object)
+	}
 	// Lock the object before reading.
 	nsMutex.RLock(bucket, object)
 	defer nsMutex.RUnlock(bucket, object)

